### PR TITLE
Change types to not assume an arch

### PR DESCRIPTION
**Pull request by Reflejo**
Some of these types were assuming that `size_t == uint64_t` or that `uint64_t == unsigned long` but that's not universally true, some BSDs such as Darwin define uint64_t as unsigned long long which warns on literals that are defined as `UL`.

Additionally there was also some assumptions made on the pch as to forward declaration. So I explicitly added some needed ones.

### CODE REVIEW

File: source/common/http/access_log/access_log_impl.cc [hunk 1]
```diff
@@ -99,7 +99,7 @@ bool RuntimeFilter::evaluate(const RequestInfo&, const HeaderMap& request_header
   const HeaderEntry* uuid = request_header.RequestId();
   uint16_t sampled_value;
   if (uuid && UuidUtils::uuidModBy(uuid->value().c_str(), sampled_value, 100)) {
-    uint64_t runtime_value = std::min(runtime_.snapshot().getInteger(runtime_key_, 0), 100UL);
+    auto runtime_value = std::min<uint64_t>(runtime_.snapshot().getInteger(runtime_key_, 0), 100);
```
Comment (mattklein123) [inline source/common/http/access_log/access_log_impl.cc]: can we keep uint64_t as the var type (prefer to not use auto outside of for loops for clarity)

File: source/common/http/http2/codec_impl.cc [hunk 2]
```diff
@@ -114,13 +114,14 @@ void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
 }
 
 ssize_t ConnectionImpl::StreamImpl::onDataSourceRead(size_t length, uint32_t* data_flags) {
-  if (pending_send_data_.length() == 0 && !local_end_stream_) {
+  size_t pending_send_data_length = pending_send_data_.length();
```
Comment (mattklein123) [inline source/common/http/http2/codec_impl.cc]: uint64_t: length() returns that and we are trying to make sure it is 64-bit unsigned (prefer to not use size_t for the reasons you are trying to fix).

File: source/common/memory/stats.cc [hunk 3]
```diff
@@ -7,13 +7,13 @@
 namespace Memory {
 
 uint64_t Stats::totalCurrentlyAllocated() {
-  uint64_t value = 0;
+  size_t value = 0;
```
Comment (mattklein123) [inline source/common/memory/stats.cc]: Sorry are these changes required because MallocExtension takes size_t as param?

Comment (Reflejo) [inline source/common/memory/stats.cc]: yep, since the constraint is on `GetNumericProperty` I think is better to be explicit here for clarity if that makes sense; also this of course warns on Darwin since `__SIZE_TYPE__` is `unsigned long` and `uint64_t` is `unsigned long long`

File: source/common/http/http2/codec_impl.cc (hunk 2 above)
Comment (Reflejo) [inline source/common/http/http2/codec_impl.cc]: Ah yeah my bad, I followed the wrong `length()` and thought it was `size_t`

File: source/common/http/access_log/access_log_impl.cc (hunk 1 above)
Comment (Reflejo) [inline source/common/http/access_log/access_log_impl.cc]: 👌

File: source/common/http/http2/codec_impl.cc (hunk 2 above)
Comment (Reflejo) [inline source/common/http/http2/codec_impl.cc]: Honestly I think we should change this signature to take `uint64_t` instead; does that makes sense to you? I'm pushing that change instead.

Comment (Reflejo) [inline source/common/http/http2/codec_impl.cc]: done

**Comment by mattklein123** [comment]
@Reflejo please fix format and merge master
